jupyter-extension/src/plugins/mount/mountDrive.ts

```diff
@@ -121,45 +133,41 @@ export class MountDrive implements Contents.IDrive {
 
     // If we don't have contents cached, then we fetch them and cache the results.
     if (localPath !== this._cache.key || !localPath) {
+      const now = Date.now();
+      this._cache = {
+        key: localPath,
+        now,
+        contents: [],
+        filteredContents: [],
+      };
       this._loading.emit(true);
       const getOptions = {
         ...options,
         number: PAGINATION_NUMBER,
         content,
       };
       const response = await this._get(url, getOptions);
+      this._cache.contents = response.content;
+      this.filterContents();
+      this.resetContentsNode();
       this._loading.emit(false);
-      this._model.page = 1;
-      const now = Date.now();
-      this._cache = {key: localPath, now, contents: response.content};
       this._fetchNextPage(response, now, url, getOptions);
     }
 
-    // Filter contents if filter defined
-    const filter = this.getFilter();
-    const contents = !filter
-      ? this._cache.contents
-      : this._cache.contents.filter((content: Contents.IModel) => {
-          return content.name.toLowerCase().includes(filter);
-        });
-
-    // Reset page to 1 if the filter has changed and update the previousFilter for this check
-    // on the next call of this method.
-    if (filter !== this._previousFilter) {
-      this._model.page = 1;
+    const newFilter = this.getFilter();
+    if (newFilter !== this._previousFilter) {
+      this.resetContentsNode();
+      this.filterContents();
     }
-    this._previousFilter = filter;
+    this._previousFilter = newFilter;
 
-    // Update max page and pagination UI after contents have changed.
-    this._model.max_page = Math.ceil(contents.length / MAX_NUM_CONTENTS_PAGE);
-    this._pagingUpdate();
+    this.setupScrollingHandler();
 
+    const {start} = this.getContentsStart();
+    const {end} = this.getContentsEnd();
     return {
       ...shallowResponse,
-      content: contents.slice(
-        (this._model.page - 1) * MAX_NUM_CONTENTS_PAGE,
-        this._model.page * MAX_NUM_CONTENTS_PAGE,
-      ),
+      content: this._cache.filteredContents.slice(start, end),
```

Comment:
Hmm I noticed one issue with refreshing the page to a directory with a lot of files (as opposed to navigating to that directory from root/parent directory). Seems like it loads some sort of cache from the previous session which is then is overwrote once the first _get call finishes causing the scrollbar to reset to zero. 

It still works, but is a little odd from a UX perspective. The issue also existed before this PR. Still worth reviewing this as is since I can work on that issue and submit a separate PR for the fix (or follow up commit).

Reply:
Fixed this issue in the next commit I am pushing!